jquery.flot.hover.js

```diff
@@ -0,0 +1,303 @@
+                clickable: true
+            }
+        },
+        plot, highlights = [];
+
+    function init(plot) {
+        plot.highlight = highlight;
+        plot.unhighlight = unhighlight;
+        plot.hooks.processOptions.push(initHover);
+    }
+
+    function initHover(flot, options) {
+        plot = flot;
+
+        function bindEvents(plot, eventHolder) {
+            var o = plot.getOptions();
+
+            if (o.grid.hoverable || o.grid.clickable) {
+                eventHolder[0].addEventListener('touchevent', triggerCleanupEvent, false);
+                eventHolder[0].addEventListener('tap', tap.generatePlothoverEvent, false);
+            }
+
+            if (options.grid.clickable) {
+                eventHolder.click(onClick);
+            }
+
+            if (options.grid.hoverable) {
+                eventHolder.mousemove(onMouseMove);
+
+                // Use bind, rather than .mouseleave, because we officially
+                // still support jQuery 1.2.6, which doesn't define a shortcut
+                // for mouseenter or mouseleave.  This was a bug/oversight that
+                // was fixed somewhere around 1.3.x.  We can return to using
+                // .mouseleave when we drop support for 1.2.6.
+
+                eventHolder.bind("mouseleave", onMouseLeave);
+            }
+        }
+
+        function shutdown(plot, eventHolder) {
+            eventHolder[0].removeEventListener('tap', tap.generatePlothoverEvent);
+            eventHolder[0].addEventListener('tap', triggerCleanupEvent);
+            eventHolder.unbind("mousemove", onMouseMove);
+            eventHolder.unbind("mouseleave", onMouseLeave);
+            eventHolder.unbind("click", onClick);
+            highlights = [];
+        }
+
+        var tap = {
+            generatePlothoverEvent: function (e) {
+                var o = plot.getOptions(),
+                    series = plot.getData(),
+                    newEvent = new CustomEvent('mouseevent');
+
+                //transform from touch event to mouse event format
+                newEvent.pageX = e.detail.changedTouches[0].pageX;
+                newEvent.pageY = e.detail.changedTouches[0].pageY;
+
+                if (o.grid.hoverable) {
+                    triggerClickHoverEvent('plothover', newEvent,
+                        function(i) {
+                            return series[i]['hoverable'] !== false;
+                        }, 30);
+                }
+                return false;
+            }
+        };
+
+        if (options.grid.hoverable || options.grid.clickable) {
+            plot.hooks.bindEvents.push(bindEvents);
+            plot.hooks.shutdown.push(shutdown);
+            plot.hooks.drawOverlay.push(drawOverlay);
+        }
+
+        var triggerCleanupEvent = function(e) {
+            plot.unhighlight();
+            plot.getPlaceholder().trigger(new CustomEvent('plothovercleanup', { detail: e }));
+        }
+    }
+
+    function onMouseMove(e) {
```

Comment:
For consistency with the touch event handler(s), these mouse related handlers should be grouped in their own object, or the touch event handler(s) moved outside their {}.